src/assets/src/components/common.tsx

```diff
@@ -274,38 +271,47 @@ interface SingleInputFormShowRemainingProps extends StatelessSingleInputFormProp
 }
 
 export const SingleInputFormShowRemaining: React.FC<SingleInputFormShowRemainingProps> = (props) => {
-    const inputRef = createRef<HTMLTextAreaElement>();
-    useEffect(() => {
-        if (!props.autofocus) return;
-        inputRef.current!.focus();
-    }, []);
-    const submit = (e: React.FormEvent<HTMLFormElement>) => {
-        e.preventDefault();
-        props.onSubmit(props.value);
-        props.onChangeValue("");
+    const [remaining, setRemaining] = useState(props.maxLength - props.value.length as number)
+    const [isInvalid, setIsInvalid] = useState(false as boolean)
+    const buttonClass = `btn btn-${props.buttonType} remaining-controls` as string
+
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault()
+        if (!isInvalid) {
+            props.onSubmit(props.value)
+            props.onChangeValue('')
+        }
     }
-    const buttonClass = "btn btn-" + props.buttonType + " remaining-controls";
-    const [remaining, setRemaining] = useState(props.maxLength - props.value.length);
-    const handleChange = (newValue:string) => {
+    const handleChange = (newValue: string) => {
         props.onChangeValue(newValue);
+        if (newValue.length <= props.maxLength) { setIsInvalid(false) } else { setIsInvalid(true) }
         setRemaining(props.maxLength - newValue.length)
     }
+
+    const charsRemaining = (remaining > 0) ? remaining : 0 as number
+    const charsOver = (remaining < 0) ? ` (${remaining * -1} over limit)` : '' as string
+    const feedbackText = `Remaining characters: ${charsRemaining}/${props.maxLength}` + charsOver as string
```

Comment:
I'm happy to change this based on feedback from @mfldavidson or @teshsopher. I just found the use of only numbers and negatives to be a little jarring. This code might also be simplified a bit if I changed the `remaining` `useState` hook, but I opted to not mess with things even more.

Reply:
No need for the type assertions here, but otherwise I think it's good.

Reply:
Removed the type assertions.

Reply:
Seems like I got the okay from the team on the text changes.